core/src/test/java/org/elasticsearch/index/query/GeoDataGenerator.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.elasticsearch.common.geo.GeoPoint;
+import org.elasticsearch.common.geo.GeoUtils;
+import org.elasticsearch.test.ElasticsearchTestCase;
+
+/** Helper class for generating geo related data. */
+public class GeoDataGenerator {
+
+    /** Generate a valid random geo point. */
+    public static GeoPoint randomGeoPoint() {
+        double lat = ElasticsearchTestCase.randomDoubleBetween(GeoUtils.MIN_LAT, GeoUtils.MAX_LAT, true);
+        double lon = ElasticsearchTestCase.randomDoubleBetween(GeoUtils.MIN_LON, GeoUtils.MAX_LON, true);
+        return new GeoPoint(lat, lon);
+    }
+
+    /** Generate a valid random bounding box. */
+    public static BoundingBox randomBoundingBox() {
```

Comment:
This is a reminder from cherry-picking a helper class I needed over in the Bounding Box refactoring here - it's no longer needed there, so I'll remove it here as well.